DOC: improve interface names in docstring examples

hagenw opened the pull request:
This pull request ensures that the docstring examples of all interfaces use always the word `interface` as variable name for an interface. Before,it could have been a mixture of `versioned`, `maven`, ..., depending which methods were inherited and which not  (as first discussed in https://github.com/audeering/audbackend/pull/194).

The examples for `audbackend.interface.Unversioned` now are:

![image](https://github.com/audeering/audbackend/assets/173624/f3cac314-9e0d-410e-ae7a-c6b46d2863a4)

frankenjoe commented on audbackend/core/errors.py:
```diff
@@ -4,14 +4,24 @@ class BackendError(Exception):
+        >>> backend = FileSystem("host", "repo")
         >>> try:
-        ...     unversioned.checksum("/does/not/exist")
+        ...     interface = audbackend.interface.Unversioned(backend)
```
It feels a bit weird we use `audbackend.interface.Unversioned` vs `FileSystem`. So maybe we should also make `audbackend.interface.Unversioned` and ``audbackend.interface.Versioned` available as `Unversioned` and `Versioned`?

hagenw replied:
That was indeed not nice.
I updated the code and now always use `audbackend.backend.FileSystem` in the visible docstring examples.

hagenw replied:
If we like, we could also remove `DoctestFileSystem` and skip the doctests for `date` and `owner`. But I'm also fine with the current solution.

hagenw commented on audbackend/core/interface/base.py:
```diff
@@ -18,6 +18,16 @@ class Base:
+        >>> FileSystem.create("host", "repo")
+
+    Examples:
+        >>> interface = Base(FileSystem("host", "repo"))
```
I added it as its methods/attributes also have examples. And those we added as they are used automatically in other interfaces, in which we don't implement those methods/attributes.

frankenjoe replied:
Yes, the examples in the methods we can keep. I am just not sure about this specific one.

hagenw replied:
OK, I'm fine with that and removed it.

frankenjoe commented:
I like how you rewrote the examples in the docstrings by locally creating objects and files. It simplifies the code and also has the advantage that we don't share objects / files across the tests. The latter was the reason to use rather cryptic file names like `f.ext` to avoid long lines since e.g. `ls()` would show all files that we needed throughout the tests. Now we can specifically control which files a doctest will see. I would therefore recommend to change `f.ext` to `file.ext` again and on that account create less files with `ls()`.

hagenw commented:
Good idea. I first updated only the `Unversioned` examples (also in the screenshot). If you agree with those changes, I will update the other interfaces as well.

frankenjoe commented:
> Good idea. I first updated only the `Unversioned` examples (also in the screenshot). If you agree with those changes, I will update the other interfaces as well.

Looks good. Here I would change to `/file.txt`, though:

![image](https://github.com/audeering/audbackend/assets/10383417/53431965-20d5-475b-864c-787ad43ce1ad)

Btw: the indent of the `date` and `owner` example looks wrong in my editor. Not sure why `black` doesn't complain.

hagenw commented:
Good catch, I fixed that and updated the other interfaces now as well.

hagenw commented:
Interesting, there was indeed too much space. I fixed it now.

frankenjoe commented:
I wonder if we should take it even one step further and adjust the doctests that they run completely isolated, e.g.:

```python
            >>> import audeer
            >>> audbackend.backend.FileSystem.create("host", "repo")
            >>> backend = audbackend.backend.FileSystem("host", "repo")
            >>> interface = Unversioned(backend)
            >>> interface.put_file(audeer.touch("file.txt"), "/file.txt")

        Examples:
            >>> interface.exists("/file.txt")
            True

        ..
            >>> audbackend.backend.FileSystem.delete("host", "repo")
```

Now the example does not depend on any previously executed example. Would be the cleanest way to organize the doctests. Or you think it's an overkill?

hagenw commented:
The problem when not depending on the code executed before is that we would need to list all the steps we do to the user, otherwise there is no clue why we are getting the result. At the moment, we avoid this by relying on the examples run before.

frankenjoe commented:
When I read an API I usually look only at the current method I'm interested in and I don't want to look at all previous methods to understand what is needed to run the example. Usually that also fine since I only want to see how I can call that particular function. And if I am really interested in the full example code, I can look into the source code and have everything in one place.

But I am also fine with the current solution, of course.

hagenw commented:
I have updated the docstring examples for `Unversioned` (see screenshot in description) now. I also adjusted `audbackend/core/conftest.py` accordingly, this is the reason why the test now fail for the other docstring examples.

frankenjoe commented:
Looks great now!

Only here I would use `/file.txt` like with the other examples:

![image](https://github.com/audeering/audbackend/assets/10383417/43d65c6b-8b25-4871-8d96-d3674d1febd7)

hagenw commented:
I integrated your suggestion, and updated all other docstring examples now as well.
